### ISSUE TITLE
can't access to my admin ....

### ISSUE THREAD

**Issue by TOMER628** [nicolargo/varnish-nginx-wordpress#19]
hi ! 

i like your job it's awesome cache.
but i have a problem to enter my admin area:

mywebsite.com/s-login

or another website:

mywebsite.com/tomer-login

he said : 
" You must log in to access the admin area. "
and i am the admin , i don't how to fix that , before this Code of Default.vcl i was free access to my admin area without problem..

what i need to do to fix that ?

regards,
Tomer

**Comment by Oyabi** [comment]
Hi,
i default.vcl change :

    # Did not cache the admin and login pages
    if (req.url ~ "/wp-(login|admin)") {
        return (pass);
    }
By : 

    # Did not cache the admin and login pages
    if (req.url ~ "/(s-login|tomer-login|wp-login|wp-admin)") {
        return (pass);
    }
You can remove wp-login and wp-admin if you don't use them.
Regards.

**Comment by TOMER628** [comment]
wow you fast !
i tried that now but still same error

**Comment by Oyabi** [comment]
Have you custom the file varnish4-wordpress ?
I remember that I have similary error and I think these lines resolve it : 

    # Remove the wp-settings-1 cookie
    set req.http.Cookie = regsuball(req.http.Cookie, "wp-settings-1=[^;]+(; )?", "");
    # Remove the wp-settings-time-1 cookie
    set req.http.Cookie = regsuball(req.http.Cookie, "wp-settings-time-1=[^;]+(; )?", "");
    # Remove the wp test cookie
    set req.http.Cookie = regsuball(req.http.Cookie, "wordpress_test_cookie=[^;]+(; )?", "");

I don't remember exactly which repair your problem but try to search all "cookie" occurence in varnish4-wordpress. =)

**Comment by TOMER628** [comment]
so just add this line to code?

**Comment by Oyabi** [comment]
yep, look at this file ;) 
https://github.com/nicolargo/varnish-nginx-wordpress/blob/master/varnish/varnish4-wordpress
Maybe anothers lines must be add, but there all in example file ;)

**Comment by TOMER628** [comment]
this exactly the code i copied ...
and also i added this line into file eight now ' and still error with cookies

**Comment by TOMER628** [comment]
by the way this:

# Only allow purging from specific IPs
acl purge {
    "localhost";
    "127.0.0.1";
}

i did not change the ip (127.0.0.1) this ok right ?

**Comment by Oyabi** [comment]
Sorry i haven't read all my file earlier.
You must also ahve theses lines in sub vcl_backend_response : 

   # Only allow cookies to be set if we're in admin area
   if (beresp.http.Set-Cookie && bereq.url !~ "^/wp-(login|admin)") {
      unset beresp.http.Set-Cookie;
   }
For your configuration you must change it to : 

 # Only allow cookies to be set if we're in admin area
   if (beresp.http.Set-Cookie && bereq.url !~ "^/(s-login|tomer-login|wp-login|wp-admin)") {
      unset beresp.http.Set-Cookie;
   }
I hope your error is gone.

**Comment by TOMER628** [comment]
yes !!!!!!
you did it !!!! 

Great !

thank you so much 

i think maybe you need to add this to your code ..

**Comment by Oyabi** [comment]
It's in the file, but you must adapt it to correpond to your custom install.
By default you can access wordpress admin with /wp-admin address.
You modify it for use (s-login|tomer-login) so you must add this change in config file ;)
Have a nice day.

**Comment by TOMER628** [comment]
thank you so much i like your code it make my Wordpress fastest

**Comment by TOMER628** [comment]
hi again..
i forgot to ask:
i need also this plugin (or your code do that automaticlly ?
LINK:    https://wordpress.org/plugins/varnish-http-purge/

**Comment by nicolargo** [comment]
No, you had to use this plugin in order to purge the cache (for exemple after an WP plugins update).

**Comment by TOMER628** [comment]
SO the answer is not, right ?

**Comment by nicolargo** [comment]
Yes the answer is... not.

**Comment by TOMER628** [comment]
THANKS
